src/pull_request.ts

```diff
@@ -80,26 +131,32 @@ export async function handlePullRequest() {
   });
   info(`generated pull request summary: ${summary.title}`);
 
-  // Update PR title and description
-  //   await octokit.rest.pulls.update({
-  //     ...context.repo,
-  //     pull_number: pull_request.number,
-  //     title: summary.title,
-  //     body: summary.description,
-  //   });
-  //   info(`updated pull request title and description`);
+  // Update PR title if @presubmitai is mentioned in the title
+  if (pull_request.title.includes("@presubmitai")) {
+    info(`title contains @presubmitai, so generating a new title`);
+    await octokit.rest.pulls.update({
+      ...context.repo,
+      pull_number: pull_request.number,
+      title: summary.title,
+      // body: summary.description,
+    });
+  }
```

Comment:
Consider adding validation for the new title before updating:
1. Remove the commented out code for better maintainability
2. Add validation for the new title to prevent setting empty or invalid titles
```typescript
if (pull_request.title.includes("@presubmitai") && summary.title?.trim()) {
  info(`title contains @presubmitai, so generating a new title`);
  await octokit.rest.pulls.update({
    ...context.repo,
    pull_number: pull_request.number,
    title: summary.title
  });
} else if (pull_request.title.includes("@presubmitai")) {
  warning('Generated title is empty or invalid, skipping title update');
}
```